clang/lib/Sema/SemaDecl.cpp

```diff
@@ -14608,6 +14608,10 @@ void Sema::CheckCompleteVariableDeclaration(VarDecl *var) {
   std::optional<bool> CacheHasConstInit;
   const Expr *CacheCulprit = nullptr;
   auto checkConstInit = [&]() mutable {
+    const Expr *Init = var->getInit();
+    if (Init->isInstantiationDependent())
```

Comment:
So i did a quick audit of the uses of this, and I think this will cause a crash later. 

line 14724 does `checkForConstantInitialization` if this is true.  

I'm not sure how to get a repro for that, but I think that ends up being wrong. Though 14734 is going to cause a similar problem...

Reply:
Oh wait, 14706 seems to protect against that, it has the value-dependent check.  So NVM.